rct229/rulesets/ashrae9012019/ruleset_functions/compare_swh_dist_systems_and_components.py

```diff
@@ -0,0 +1,132 @@
+from rct229.rulesets.ashrae9012019.data_fns.extra_schema_fns import (
+    EXTRA_SCHEMA,
+    compare_context_pair,
+)
+from rct229.rulesets.ashrae9012019.ruleset_functions.get_swh_equipment_associated_with_each_swh_distriubtion_system import (
+    get_swh_equipment_associated_with_each_swh_distribution_system,
+)
+from rct229.utils.assertions import assert_
+from rct229.utils.utility_functions import (
+    find_exactly_one_pump,
+    find_exactly_one_service_water_heating_distribution_system,
+    find_exactly_one_service_water_heating_equipment,
+)
+
+
+def compare_swh_dist_systems_and_components(
+    rmd1: dict, rmd2: dict, compare_context_str: str, swh_distribution_id: str
```

Comment:
Could @weilixu let me know what ```compare_context_str``` argument indicates?

Reply:
`compare_context_str` is the flag string in the extra schema:

https://github.com/pnnl/ruleset-checking-tool/blob/develop/rct229/rulesets/ashrae9012019/data/ASHRAE229.9012019.extra.schema.json#L94-L96

Examples such as `AppG P_RMD Equals U_RMD`.

Reply:
Good point. Thanks!